packages/orbit-components/src/Tag/index.tsx

```diff
@@ -80,35 +79,80 @@ const Tag = React.forwardRef<HTMLDivElement, Props>(
           ],
         )}
         data-test={dataTest}
-        id={id}
-        ref={ref}
-        onClick={onClick}
-        // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
-        tabIndex={(onClick || onRemove) && 0}
-        role={onClick || onRemove ? "button" : undefined}
-        onKeyDown={ev => buttonClickEmulation(ev, onClick)}
       >
-        {iconLeft && (
-          <div className="pe-200 [&_svg]:size-icon-small flex flex-row items-center justify-center">
-            {iconLeft}
-          </div>
-        )}
-        {children}
+        <div
+          className={cx(
+            "orbit-tag flex items-center",
+            "focus:rounded-150",
+            onRemove ? "ps-200 py-200 pe-100 rounded-l-150" : "p-200 rounded-150",
+            selected && [
+              dateTag
+                ? [!!(onClick || onRemove) && "focus-visible:bg-ink-light-active focus:z-default"]
+                : [
+                    !!(onClick || onRemove) &&
+                      " focus-visible:bg-blue-normal-hover focus:z-default",
+                  ],
+            ],
+            type === TYPES.NEUTRAL && [
+              !selected && [
+                !!(onClick || onRemove) && "focus-visible:bg-cloud-normal-hover focus:z-default",
+              ],
+            ],
+            type === TYPES.COLORED && [
+              !selected && [
+                !!(onClick || onRemove) && "focus-visible:bg-blue-light-hover focus:z-default",
+              ],
+            ],
+          )}
+          id={id}
+          ref={ref}
+          {...((onClick || onRemove) && {
+            role: "button",
+            tabIndex: 0,
+            onClick,
+            onKeyDown: ev => buttonClickEmulation(ev, onClick),
+          })}
```

Comment:
This silences the linter because it can't analyze this part of the code 😄

But actually… looking at this, I noticed a problem with our current implementation.

We are saying that any tag with `onRemove` is rendered as a button. But we clearly have the dismiss icon separated from the "content", in two separate elements…

So why is a Tag with `onRemove` but without `onClick` focusable, clickable, and all, if only the dismiss part is clickable?

You can see this on the default story, the Tag that says "removable". I don't think it is working correctly.

Reply:
aha! good point.

Reply:
Fixed, I also did TW classNames cleanup. Regarding these "spread attributes", I was inspired by changes in `SectionHeader` file. @DSil